File: backend/assetflow-notification-service/src/main/java/com/assetflow/notification/response/ApiResponse.java
package com.assetflow.notification.response;
import java.time.Instant;
public record ApiResponse<T>(boolean success,T data,String message,Instant timestamp){ public static <T> ApiResponse<T> ok(T d){return new ApiResponse<>(true,d,null,Instant.now());} public static <T> ApiResponse<T> error(String m){return new ApiResponse<>(false,null,m,Instant.now());}}

File: dotNet-backend/AssetService/DTOs/ApiResponse.cs
namespace AssetService.DTOs
{
    public class ApiResponse<T>
    {
        public bool Success { get; set; }
        public string Message { get; set; } = string.Empty;
        public T? Data { get; set; }
        public long Timestamp { get; set; }

        public static ApiResponse<T> Ok(T data)
        {
            return new ApiResponse<T>
            {
                Success = true,
                Message = "Success",
                Data = data,
                Timestamp = DateTimeOffset.UtcNow.ToUnixTimeMilliseconds()
            };
        }

        public static ApiResponse<T> Message(string message)
        {
            return new ApiResponse<T>
            {
                Success = true,
                Message = message,
                Timestamp = DateTimeOffset.UtcNow.ToUnixTimeMilliseconds()
            };
        }

        public static ApiResponse<T> Error(string message)
        {
            return new ApiResponse<T>
            {
                Success = false,
                Message = message,
                Timestamp = DateTimeOffset.UtcNow.ToUnixTimeMilliseconds()
            };
        }
    }
}
